### DISCUSSION TITLE
time displacement

### DISCUSSION THREAD

**Discussion by FerruGit** [dbuezas/lovelace-plotly-graph-card#382] [Q&A]
Hello,

A sensor (LD2450) provides data (x(t),y(t)) which I display in a graph (x on the x-axis, y on the y-axis).
This displays the trajectory of the person detected. I display it over a period of 20 minutes (hours_to_show: 20m). If I display it over a longer period the graph becomes too busy.
I’d like to be able to shift this period in time with a horizontal bar or two buttons (< >) to move forwards or backwards in time.

Do you have any ideas on how to do this?

Thank you in advance.

**Comment by dbuezas** [comment]
This is tricky, because the x axis is used to determine the visible range and that's what's fetched from HA when you scroll/zoom
I even have a TODO there https://github.com/dbuezas/lovelace-plotly-graph-card/blob/master/src/plotly-graph-card.ts#L256

Maybe you could tell plotly to use the y and z axes to plot, then hide the x axis and finally add the buttons in the readme to change the current time range.
 
If you get this to work, please come back and tell us! :)
